app/lib/tariff_synchronizer.rb

```diff
@@ -223,46 +223,27 @@ def rollback(rollback_date, keep: false)
 
   def rollback_cds(rollback_date, keep: false)
     Rails.autoloaders.main.eager_load
+
     TradeTariffBackend.with_redis_lock do
       date = Date.parse(rollback_date.to_s)
 
-      (date..Time.zone.today).to_a.reverse.each do |date_for_rollback|
-        Sequel::Model.db.transaction do
-          if keep
-            TariffSynchronizer::CdsUpdate.applied_or_failed.where { issue_date > date_for_rollback }.each do |cds_update|
-              # Delete actual data
-              oplog_based_models.each do |model|
-                model.operation_klass.where('filename = ?', cds_update.filename).delete
-              end
-
-              instrument('rollback_update.tariff_synchronizer',
-                         update_type: :cds,
-                         filename: cds_update.filename)
-
-              cds_update.mark_as_pending
-              cds_update.clear_applied_at
+      updates = TariffSynchronizer::CdsUpdate.applied_or_failed.where { issue_date > date }
```

Comment:
Yeah, this will leave pending updates in tact. I actually don't think there's much of a loss in just using the same dataset (e.g. all states) in both cases since we're always executing the same number of queries and marking as pending, clearing the applied at and flushing out the cds errors doesn't really have an impact on a pending update anyway.

Reply:
```ruby
updates = TariffSynchronizer::CdsUpdate.where { issue_date > date }
```

Reply:
Yeh, agreed it'll work without the scope. The code will try to remove data that doesn't exist but that shouldn't cause any difficulties.